features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java

```diff
@@ -508,4 +522,63 @@ public void reloadDaemonConfigForLocation(final IEvent e) {
+            final OnmsOutage onmsOutage = new OnmsOutage(onmsEvent.getEventCreateTime(), onmsEvent, service);
+            onmsOutage.setPerspective(perspective);
+            outageDao.save(onmsOutage);
+
+            final Event outageEvent = new EventBuilder(EventConstants.OUTAGE_CREATED_EVENT_UEI, "RemotePollerd")
+                    .setNodeid(onmsEvent.getNodeId())
+                    .setService(service.getServiceName())
+                    .setTime(onmsEvent.getEventCreateTime())
+                    .setParam("perspective", perspective.getLocationName())
+                    .getEvent();
+            eventForwarder.sendNow(outageEvent);
+        } else {
+            LOG.warn("Received incomplete {} event: {}", EventConstants.REMOTE_NODE_LOST_SERVICE_UEI, e);
+        }
+    }
+
+    @EventHandler(uei = EventConstants.REMOTE_NODE_REGAINED_SERVICE_UEI)
+    public void handleRemoteNodeGainedService(final IEvent e) {
+        if (e.hasNodeid() && e.getInterfaceAddress() != null && e.getService() != null && e.getParm("perspective") != null) {
+            final OnmsEvent onmsEvent = eventDao.get(e.getDbid());
+            final OnmsMonitoredService service = monSvcDao.get(onmsEvent.getNodeId(), onmsEvent.getIpAddr(), onmsEvent.getServiceType().getId());
+            final OnmsMonitoringLocation perspective = monitoringLocationDao.get(e.getParm("perspective").getValue().getContent());
+
+            final Criteria criteria = new CriteriaBuilder(OnmsOutage.class)
+                    .eq("perspective", perspective)
+                    .isNull("serviceRegainedEvent")
+                    .isNull("ifRegainedService")
+                    .eq("monitoredService", service).toCriteria();
+
+            final List<OnmsOutage> onmsOutages = outageDao.findMatching(criteria);
+
+            if (onmsOutages.size() == 1) {
+                final OnmsOutage onmsOutage = onmsOutages.get(0);
+                onmsOutage.setIfRegainedService(onmsEvent.getEventCreateTime());
+                onmsOutage.setServiceRegainedEvent(onmsEvent);
+                outageDao.update(onmsOutage);
+
+                final Event outageEvent = new EventBuilder(EventConstants.OUTAGE_RESOLVED_EVENT_UEI, "RemotePollerd")
+                        .setNodeid(onmsEvent.getNodeId())
+                        .setService(service.getServiceName())
+                        .setTime(onmsEvent.getEventCreateTime())
+                        .setParam("perspective", perspective.getLocationName())
+                        .getEvent();
+                eventForwarder.sendNow(outageEvent);
+            } else {
+                LOG.warn("Found more than one outages for {} event: {}", EventConstants.REMOTE_NODE_REGAINED_SERVICE_UEI, e);
+                return;
+            }
+        } else {
+            LOG.warn("Received incomplete {} event: {}", EventConstants.REMOTE_NODE_REGAINED_SERVICE_UEI, e);
+        }
+    }
```

Comment:
It seems a bit strange to react to an event which is created by the same class. Any reason why we create the outage by reacting to the event instead of creating it directly when we create the event? This would safe a lot of event parameter handling and checking.

Reply:
The problem is that the service lost event is needed to generate an outage. But if I create the event and send it to the eventbus it is not necessarily persisted immediately. Only when I consume the event I can be sure that it is present in the database. If you look at the class QueryManagerDaoImpl you will see the same behavior, but in my opinion it is realizied in a much more complicated way. The outage is persisted there without the event and after the event is received the outage is updated with the event’s id. In my opinion this is too heavy and unnecessary, since I cannot imagine a case in which I need an persisted outage before the original service lost event was properly processed by OpenNMS.

Reply:
Would be handy if we could do something like:
```
CompletableFuture<Event> future =  eventForwarder.sendNow(outageEvent);
Event eventWithDbId = future.get();
```